### ISSUE TITLE
Dropbox I/O Timeout

### ISSUE THREAD

**Issue by buggles333** [ncw/rclone#490]
**Linux Ver :** Debian 3.16.0-4amd64
**Rclone Ver :** rclone 1.28
**Cloud Storage :** Dropbox

Sometimes when I run this command...

`rclone sync --exclude-from rclone_exclude.txt /mnt/drive1/public DropBox:public`

I get the following error....

```
2016/05/19 02:00:02 Dropbox root 'public': Building file list
2016/05/19 02:00:08 Dropbox root 'public': Waiting for checks to finish
2016/05/19 02:00:08 Dropbox root 'public': Waiting for transfers to finish
2016/05/19 02:00:18 public/myfile.tif: Failed to copy: Upload failed: Post https://api-content.dropbox.com/1/chunked_upload: dial tcp: lookup api-content.dropbox.com on 192.168.1.1:53: read udp 192.168.1.120:48052->192.168.1.1:53: i/o timeout
2016/05/19 02:00:18 public/spread.xls: Failed to copy: Upload failed: Post https://api-content.dropbox.com/1/chunked_upload: dial tcp: lookup api-content.dropbox.com on 192.168.1.1:53: read udp 192.168.1.120:48052->192.168.1.1:53: i/o timeout
2016/05/19 02:00:19 Attempt 1/3 failed with 2 errors
2016/05/19 02:00:19 Dropbox root 'public': Building file list
2016/05/19 02:00:24 Dropbox root 'public': Waiting for checks to finish
2016/05/19 02:00:24 Dropbox root 'public': Waiting for transfers to finish
2016/05/19 02:01:02 

Transferred:      8814080 Bytes ( 141.07 kByte/s)
Errors:                 0
Checks:             12441
Transferred:            3
Elapsed time:        1m1s
Transferring:
 * ...myfile.tif: 34% done. avg:  243.6, cur:  259.2 kByte/s. ETA: 59s


Transferred:     24522516 Bytes ( 218.50 kByte/s)
Errors:                 0
Checks:             12441
Transferred:            4
Elapsed time:     1m49.6s
```

If I am reading it correctly it looks like it succeeds in transferring the files the second time around.

Am I correct?  And also, could somebody explain what the i/o timeout is likely to be caused by?

**Comment by ncw** [comment]
I'm afraid timeouts are a fact of life with cloud storage providers.  That is why rclone retries things and you can see that since it said "Errors: 0" at the end, all was successful.

What appeared to happen is that it transferred the file, got a timeout message, then on the retry it tried again.

If this happens regularly then I can investigate further, but if it just a one off thing I'd say that is just life!

**Comment by buggles333** [comment]
Thanks for clearing that up, I had assumed this was the case but wanted to make sure!

Could I suggest a change of wording perhaps?  The fact that it says 4 files were transferred, I feel, is slightly confusing.

Would having a "Successful Transfers" and "Attempted transfers" readout make more sense?

**Comment by left1000** [comment]
imo, there's no need to list successful transfers and attempted transfers really. It starts off with transfers and errors, then it makes another loop and tries to convert the errors to transfers. if you're ever curious you can just parse the verbose logfile or something.

**Comment by buggles333** [comment]
It just seems a bit odd that to get the correct stats on files transferred you need to check the logfile.

In my example above it states that 4 files were transferred when in fact only 2 files were transferred.  Am I alone in thinking this seems a bit counter-intuitive?

**Comment by left1000** [comment]
Well for me at least, usually when 2 files appear on the cloud and it states that 4 files were transferred that means it made 4 uploads with a final result of 2 files.

Most often for me this means I uploaded two large files, large enough that the cloud provider mistakenly thought there was an upload error (or there really was one) and it restarted the uploads all over.

If for example I cared about how much bandwidth I was using (which I personally don't) the 4 number would be more useful than the 2 number because I used more than 2 files worth of bandwidth to get the cloud to recognize those two files.

I guess I just got used to the display how it is now, because the first time I used rclone I stared at my screen. Maybe your way would be better. I am personally used to the default rclone output being very minimal and then the -v output being very maximal.

Your suggestion sounds logical, I just personally don't see your way as more intuitive because I'm already used to the opposite. Sorry if I came off as combative.
